gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/grouping/Grouping.java

```diff
@@ -111,26 +126,50 @@
   private final List<PropertyValueAggregator> edgeAggregators;
 
   /**
+   * Stores grouping properties for vertex labels.
+   */
+  private final List<LabelGroup> vertexLabelGroups;
+
+  /**
+   * Stores grouping properties for edge labels.
+   */
+  private final List<LabelGroup> edgeLabelGroups;
+
+  /**
+   * Stores all aggregator property keys for each label.
+   */
+  private Map<String, Set<String>> labelWithAggregatorPropertyKeys;
+
+  /**
    * Creates grouping operator instance.
    *
-   * @param vertexGroupingKeys  property keys to group vertices
-   * @param useVertexLabels     group on vertex label true/false
-   * @param vertexAggregators   aggregate functions for grouped vertices
-   * @param edgeGroupingKeys    property keys to group edges
-   * @param useEdgeLabels       group on edge label true/false
-   * @param edgeAggregators     aggregate functions for grouped edges
+   * @param vertexGroupingKeys              property keys to group vertices
+   * @param useVertexLabels                 group on vertex label true/false
+   * @param vertexAggregators               aggregate functions for grouped vertices
+   * @param edgeGroupingKeys                property keys to group edges
+   * @param useEdgeLabels                   group on edge label true/false
+   * @param edgeAggregators                 aggregate functions for grouped edges
+   * @param vertexLabelGroups               stores grouping properties for vertex labels
+   * @param edgeLabelGroups                 stores grouping properties for edge labels
+   * @param labelWithAggregatorPropertyKeys stores all aggregator property keys for each label.
    */
   Grouping(
     List<String> vertexGroupingKeys, boolean useVertexLabels,
     List<PropertyValueAggregator> vertexAggregators,
     List<String> edgeGroupingKeys, boolean useEdgeLabels,
-    List<PropertyValueAggregator> edgeAggregators) {
-    this.vertexGroupingKeys = vertexGroupingKeys;
-    this.useVertexLabels    = useVertexLabels;
-    this.vertexAggregators  = vertexAggregators;
-    this.edgeGroupingKeys   = edgeGroupingKeys;
-    this.useEdgeLabels      = useEdgeLabels;
-    this.edgeAggregators    = edgeAggregators;
+    List<PropertyValueAggregator> edgeAggregators,
+    List<LabelGroup> vertexLabelGroups,
+    List<LabelGroup> edgeLabelGroups,
+    Map<String, Set<String>> labelWithAggregatorPropertyKeys) {
```

Comment:
Yes of course they can and after your other comments I thought of something like this:
`builder.addVertexLabelGroup(groupName, vertexLabel, propertyKeySet, aggregatorSet)`.